src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -56,37 +61,83 @@ public Spool(DeviceConfiguration deviceConfiguration) {
-                .findOrDefault(DEFAULT_GG_SPOOL_MAX_MESSAGE_QUEUE_SIZE_IN_BYTES,
-                        GG_SPOOL_MAX_SIZE_IN_BYTES_KEY));
-        boolean ggSpoolKeepQos0WhenOffline = Coerce.toBoolean(topics
-                .findOrDefault(DEFAULT_KEEP_Q0S_0_WHEN_OFFLINE, GG_SPOOL_KEEP_QOS_0_WHEN_OFFLINE_KEY));
-
-        logger.atInfo().kv(GG_SPOOL_STORAGE_TYPE_KEY, ggSpoolStorageType)
-                .kv(GG_SPOOL_MAX_SIZE_IN_BYTES_KEY, ggSpoolMaxMessageQueueSizeInBytes)
-                .kv(GG_SPOOL_KEEP_QOS_0_WHEN_OFFLINE_KEY, ggSpoolKeepQos0WhenOffline)
+        SpoolerStorageType spoolStorageType = Coerce.toEnum(SpoolerStorageType.class, topics
+                .findOrDefault(DEFAULT_SPOOL_STORAGE_TYPE, SPOOL_STORAGE_TYPE_KEY));
+        long spoolMaxMessageQueueSizeInBytes = Coerce.toLong(topics
+                .findOrDefault(DEFAULT_SPOOL_MAX_MESSAGE_QUEUE_SIZE_IN_BYTES,
+                        SPOOL_MAX_SIZE_IN_BYTES_KEY));
+        boolean spoolKeepQos0WhenOffline = Coerce.toBoolean(topics
+                .findOrDefault(DEFAULT_KEEP_Q0S_0_WHEN_OFFLINE, SPOOL_KEEP_QOS_0_WHEN_OFFLINE_KEY));
+        String persistenceSpoolerServiceName = Coerce.toString(topics
+                .findOrDefault(DEFAULT_GG_PERSISTENCE_SPOOL_SERVICE_NAME, PERSISTENCE_SPOOL_SERVICE_NAME_KEY));
+
+        logger.atInfo().kv(SPOOL_STORAGE_TYPE_KEY, spoolStorageType)
+                .kv(SPOOL_MAX_SIZE_IN_BYTES_KEY, spoolMaxMessageQueueSizeInBytes)
+                .kv(SPOOL_KEEP_QOS_0_WHEN_OFFLINE_KEY, spoolKeepQos0WhenOffline)
                 .log("Spooler has been configured");
 
-        this.config = SpoolerConfig.builder().storageType(ggSpoolStorageType)
-                .spoolSizeInBytes(ggSpoolMaxMessageQueueSizeInBytes)
-                .keepQos0WhenOffline(ggSpoolKeepQos0WhenOffline).build();
+        this.config = SpoolerConfig.builder().storageType(spoolStorageType)
+                .spoolSizeInBytes(spoolMaxMessageQueueSizeInBytes)
+                .keepQos0WhenOffline(spoolKeepQos0WhenOffline)
+                .persistenceSpoolServiceName(persistenceSpoolerServiceName).build();
     }
 
     /**
      * create a spooler instance.
+     *
      * @return CloudMessageSpool    spooler instance
      */
     private CloudMessageSpool setupSpooler() {
         if (config.getStorageType() == SpoolerStorageType.Memory) {
             return new InMemorySpool();
+        } else if (config.getStorageType() == SpoolerStorageType.Disk) {
+            try {
+                return getPersistenceSpoolGGService();
+            } catch (ServiceLoadException | IOException e) {
+                //log and use InMemorySpool
+                logger.atWarn()
+                        .kv(PERSISTENCE_SPOOL_SERVICE_NAME_KEY, config.getPersistenceSpoolServiceName())
+                        .cause(e).log("Persistence spool set up failed, defaulting to in-memory mode");
+                return new InMemorySpool();
```

Comment:
always create an in memory spooler so that we can fallback to it if anything fails.
if we cannot write to the disk spool, we need to fallback to in memory so that we are able to send messages

Reply:
Sounds right! Will add that.